napari/_qt/qt_viewer.py

```diff
@@ -1059,7 +1069,69 @@ def dropEvent(self, event):
             else:
                 filenames.append(url.toString())
 
-        self.viewer.open(filenames, stack=bool(shift_down))
+        # if trying to open as a stack, open with any available reader
+        if shift_down:
+            self.viewer.open(filenames, stack=bool(shift_down))
+            return
+
+        for filename in filenames:
+            # get available readers for this file from all registered plugins
```

Comment:
I think I'd like to see this reader-prompting logic extracted into a new function.  Something like

```py
for filename in filenames:
    reader = self._get_or_prompt_for_preferred_reader(filename)
    self.viewer.open(filename, plugin=reader)